File: android/app/src/main/java/com/tiagohs/cinema_history/models/tmdb/person/PersonExtraInfo.kt
package com.tiagohs.cinema_history.models.tmdb.person

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class PersonExtraInfo (

	@SerializedName("id") val id : Int,
	@SerializedName("name") val name : String,
	@SerializedName("custom_name") val customName : String,
	@SerializedName("highlight_image") val highlight_image : String,
	@SerializedName("quote") val quote : String,
	@SerializedName("profile") val profile : List<PersonProfile>,
	@SerializedName("awards") val awards : String,
	@SerializedName("videos") val videos : List<PersonVideo>
): Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Person/PersonExtraInfo.swift
//
//  PersonExtraInfo.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class PersonExtraInfo: BaseLocalModel {
    var id : Int!
    var name : String?
    var customName : String?
    var highlightImagePath : String?
    var quote : String?
    var profile : [PersonProfile]?
    var awards : String?
    var videos : [PersonVideo]?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        id                                                  <- map["id"]
        name                                                <- map["name"]
        customName                                          <- map["custom_name"]
        highlightImagePath                                  <- map["highlight_image"]
        quote                                               <- map["quote"]
        profile                                             <- map["profile"]
        awards                                              <- map["awards"]
        videos                                              <- map["videos"]
    }
}
